backend/src/main/java/dccs/academy/resources/UserResource.java

```diff
@@ -0,0 +1,41 @@
+package dccs.academy.resources;
+
+import dccs.academy.dtos.UserDto;
+import dccs.academy.services.UserService;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import java.util.List;
+
+@ApplicationScoped
+@Produces(MediaType.APPLICATION_JSON)
+@Path("/backend/users")
+public class UserResource {
+    @Inject UserService userService;
+    @GET
+    public Response searchUsers(
+            @QueryParam("firstName") String firstName,
+            @QueryParam("lastName") String lastName,
+            @QueryParam("plant") String plant,
+            @QueryParam("id") String id,
+            @QueryParam("departmentId") String departmentId) {
+        try {
+
+            firstName = (firstName == null || firstName.isEmpty()) ? "" : firstName.trim();
+            lastName = (lastName == null || lastName.isEmpty()) ? "" : lastName.trim();
+            plant = (plant == null || plant.isEmpty()) ? "" : plant;
+            Long userId = (id != null && !id.isEmpty()) ? Long.parseLong(id) : null;
```

Comment:
1. Use actual data types, instead of parsing them.
2. no code duplicates, this string check can actually be function, but here I am not sure if you actually need it. Even if you do this handling should be done in service methods.